lib/formats/dotnet-core-parser.js

```diff
@@ -1,82 +1,153 @@
+'use strict';
 var Dependency = require('../dependency');
 var debug = require('debug')('snyk');
 
+//TODO: any convention for global vars? (gFreqDeps)
+let freqDeps = [];
+
+function initFreqDepsDict() {
+  freqDeps['Microsoft.NETCore.Platforms'] = false;
+  freqDeps['Microsoft.NETCore.Targets'] = false;
+  freqDeps['System.Runtime'] = false;
+  freqDeps['System.IO'] = false;
+  freqDeps['System.Text.Encoding'] = false;
+  freqDeps['System.Threading.Tasks'] = false;
+  freqDeps['System.Reflection'] = false;
+  freqDeps['System.Globalization'] = false;
+  freqDeps.dependencies = {};  
+}
+
 function convertFromPathSyntax(path) {
-  var name = path.split('/').join('@'); // posix
+  let name = path.split('/').join('@'); // posix
   name = name.split('\\').join('@'); // windows
   return name;
 }
 
 function collectFlatList(targetObj) {
-  var names = Object.keys(targetObj);
-  var libs = names.map(function (name) {
+  const names = Object.keys(targetObj);
+  return names.map(function (name) {
     name = convertFromPathSyntax(name);
     return name;
   });
-  return libs;
 }
 
-function registerDependency(packageTree, resolvedName) {
-  var name = resolvedName.split('@')[0];
-  var version = resolvedName.split('@')[1];
-  packageTree.dependencies[name] =
-    packageTree.dependencies[name] || new Dependency(name, version);
-  debug(resolvedName);
+function buildTreeRecursive(targetDeps, depName, parent, treeDepth) {
+  if (treeDepth > 20)
+  {
+    throw new Error('The depth of the tree is too big.');
+  }
+  
+  let depResolvedName = "";
+  let originalDepKey = "";
+  
+  debug(`${treeDepth}: Looking for '${depName}'`);
+  const exists = Object.keys(targetDeps).some(function (currentDep) {
+      let currentResolvedName = convertFromPathSyntax(currentDep);
+      if(currentResolvedName.split('@')[0].toLowerCase() === depName){
+        depResolvedName = currentResolvedName;
+        originalDepKey = currentDep;
+        debug(`${treeDepth}: Found '${currentDep}'`);
+        return true;
+      }
+  });
+
+  if (!exists) {
+    debug(`Failed to find '${depName}'`);
+    return;
+  }
+
+  const depVersion = depResolvedName.split('@')[1];
+
+  parent.dependencies[depName] = parent.dependencies[depName] || new Dependency(depName, depVersion);
+
+  Object.keys(targetDeps[originalDepKey].dependencies || {}).forEach(function (currentDep)
```

Comment:
Maybe ` ... || []).forEach`? You can't iterate an object with `forEach`, need `Object.keys(obj).forEach(...)` instead.